microprofile/microprofile-fault-tolerance-2.1/src/main/java/org/wildfly/swarm/ts/microprofile/fault/tolerance/v21/retry/RetryService.java

```diff
@@ -0,0 +1,165 @@
+package org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.retry;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.enterprise.context.RequestScoped;
+
+import org.eclipse.microprofile.faulttolerance.Asynchronous;
+import org.eclipse.microprofile.faulttolerance.Fallback;
+import org.eclipse.microprofile.faulttolerance.Retry;
+import org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.MyConnection;
+import org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.MyException;
+
+@Asynchronous // all is async with fallback method
+@RequestScoped
+public class RetryService {
+
+    private AtomicInteger counter = new AtomicInteger(1);
+
+    // First fail, second pass - there will be 1 retry before success
+    @Retry(retryOn = Throwable.class)
+    @Fallback(fallbackMethod = "processFallback")
+    public CompletionStage<MyConnection> throwable_retryOnT() throws Throwable {
+        final Integer invocationNumber = counter.getAndIncrement();
```

Comment:
Why boxing, if you immediately unbox again? This should be `int`, not `Integer`: `int invocationNumber = counter.getAndIncrement();`.

Reply:
Same for all the other methods.